src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -754,5 +845,43 @@ internal void DisableSpi()
                 Firmata.DisableSpi();
             }
         }
+
+        /// <summary>
+        /// Pings the device, to get an estimate about the round-trip time.
+        /// With some Wifi setups, the round trip time may be significantly higher than desired.
+        /// </summary>
+        /// <param name="number">The number of pings to send</param>
+        /// <returns>The list of reply times. Contains a negative value for lost packets</returns>
+        public List<TimeSpan> Ping(int number)
```

Comment:
I wonder if this method should be async and take a cancellation Token, but that whould probably require _firmata.QueryFrimwareVersion to also be async.

Reply:
That's a bit difficult, because this command is not really designed for async calls, as it uses no sequence counting. So if you where to run multiple async pings in parallel the result would be undefined. This is mostly for testing the speed of a connection, you should not use it e.g. for keep-alive scenarios.